core/src/toga/fonts.py

```diff
@@ -35,10 +35,21 @@ def __init__(
         self,
         family: str,
         size: int | str,
+        *,
+        weight: str = NORMAL,
         style: str = NORMAL,
         variant: str = NORMAL,
-        weight: str = NORMAL,
     ):
+        """This class is used to represent a font when drawing on a :any:`Canvas`. For
+        all other widgets, fonts can be controlled using the style properties which are
+        linked below.
```

Comment:
I understand that Canvas is the only practical use (at least at present), but do we want to call that out specifically here? Would it be better to talk about this in the abstract?
```suggestion
        """Constructs a reference to a registered font.
        
        This class should be used when an API requires an explicit font reference
        (e.g., when constructing a text drawing instruction for a :any:`Canvas`). In 
        most cases, fonts are controlled using style properties on the widget.
```

Reply:
OK, updated.